utils.go

```diff
@@ -553,7 +579,15 @@ func ReadString(r io.Reader) (string, error) {
 		return "", fmt.Errorf("string in input was too long")
 	}
 
-	buf := make([]byte, l)
+	bufp := stringBufPool.Get().(*[]byte)
+	buf := (*bufp)[:l] // shares same backing array as pooled slice
+	defer func() {
+		// optimizes to memclr
+		for i := range buf {
+			buf[i] = 0
+		}
```

Comment:
But we don't _really_ need this, do we?

Reply:
(although I guess it likely doesn't matter)

Reply:
Good practice to clear the buffer before returning to pool. Although we're currently careful not to read past bounds so it's not strictly necessary.